stl/inc/memory

```diff
@@ -1251,11 +1253,45 @@ private:
         _Owner._Call_deleter = false;
     }
 
-    template <class _Ty0, class... _Types>
-    friend shared_ptr<_Ty0> make_shared(_Types&&... _Args);
+    template <class _Ty, class... _Types>
+#if _HAS_CXX20

```

Comment:
Thas should rather be guarded by `__cpp_lib_concepts`

Reply:
Did you mean `__cpp_concepts`? This implementation doesn't use concepts library, it uses type_traits.

Reply:
the following line has a `requires` clause so it is using concepts?

Reply:
Yes it is using concepts but not concepts library.

Reply:
So I think it should check for `__cpp_concepts` than `__cpp_lib_concepts`

Reply:
I will leave that to one of the maintainers. I am not that deep into the details of the STL

Reply:
If I'm not mistaken `__cpp_concepts` is for the language feature (`requires` and `concept`) and `__cpp_lib_concepts` would be for the contents of `<concepts>`.

Reply:
I believe that `__cpp_concepts` is correct here since you aren't using the Library concepts, but @CaseyCarter is the expert here.